packages/@o3r/localization/README.md

```diff
@@ -1,3 +1,703 @@
+
+Examples:
+
+- [simpleHeader.airline] My Airline
+
+- [simpleHeader.motto] Let's shape the future of travel
+
+## How to properly create keys?
+
+It's always good to stick to some naming conventions.
+
+For components we prefix each key by component selector to ensure cross components keys uniqueness.
+
+For example if your `SimpleHeaderPresComponent` has `o3r-simple-header-pres` selector then your keys may look like
+
+```typescript
+
+{
+  "o3r-simple-header-pres.motto": {
+    "description": "airline motto phrase",
+    "defaultValue": "Let's shape the future of travel"
+  },
+  "o3r-simple-header-pres.airline": {
+    "description": "airline name",
+    "defaultValue": "My Airline"
+  }
+}
+
+```
+
+## Intelligent fallback support
+
+The fallback hierarchy is been added to the localization service which explained in detail below. As the last option, the default fallback language picked as usual.
+
+### Scenario 1: Fallback based on **fallbackLocalesMap**, **supportedLocales** language code
+
+Incase if **fallbackLocalesMap** provided and the targeted translation language is unavailable in supported locales list, The priority goes to **fallbackLocalesMap**, to see if targeted translation language can be mapped with the fallback map configured. The second priority goes to **supportedLocales**, to map targeted translation language to the first closest possible language, the locale can be different. If none match, it will fallback to the **default language**.
+
+**Lets assume:**
+
+```typescript
+{
+  "supportedLocales": ['en-GB', 'en-US', 'fr-FR', 'ar-AR'],
+  "fallbackLocalesMap": {
+            'en-CA': 'en-US',
+            'fr-CA': 'fr-FR',
+            'de-CH': 'ar-AR',
+            'de': 'fr-FR',
+            'it': 'fr-FR',
+            'hi': 'en-GB',
+            'zh': 'en-GB'
+  },
+  "fallbackLanguage": 'ar-AR'
+}
+```
+
+**Fallback scenario's:**
+
+en-CA **fallbacks to** en-US, as direct mapping available in fallback Locale Map.
+
+de-CH **fallbacks to** ar-AR, as direct mapping available in fallback Locale Map.
+
+de-AT **fallbacks to** fr-FR, as language mapping available in fallback Locale Map.
+
+zh-CN **fallbacks to** en-GB, as language mapping available in fallback Locale Map.
+
+en-AU **fallbacks to** en-GB, as fallback locales mapping unavailable, first nearest language available in supported locales.
+
+fr-BE **fallbacks to** fr-FR, as fallback locales mapping unavailable, first nearest language available in supported locales.
+
+bn-BD **fallbacks to** ar-AR, as it is the default fallback.
+
+### Scenario 2: Fallback based on **supportedLocales** language code
+
+In case if **fallbackLocalesMap** is not provided and the targeted translation language is unavailable is supported locales list, The targeted translation language will be matched with supported locales **language** to find out the first nearest match, the **locale** can be different. If none match, it will fallback to the **default language**.
```

Comment:
- If **fallbackLocalesMap** is not provided ... 
- unavailable **in** supported locales list, the targeted ... 
- to **find the** first **closest** match (the **locale** can be different).